sdk/samples/iot/generate_certificate.ps1

```diff
@@ -4,9 +4,8 @@
 Set-StrictMode -Version 3.0
 $errorActionPreference="stop"
 
-if($PSVersionTable.PSEdition -ne "Core")
-{
- Write-Error "This script requires Powershell Core. Please install Powershell Core at aka.ms/pscore6."
```

Comment:
I used `Write-Error` in another script. Can you explain why `throw` is better? What's wrong with using `Write-Error`?

Reply:
The output stands out a little more. The "error" message with throw is a little more by itself so it easier to recognize from users.

Reply:
If another PowerShell script invoked Dane's script, then the throw would bubble up to the caller and it could then programmatically take action or just let the unhandled exception bubble up and stop its execution as well.  Not being on correct version of PS is serious enough to stop the show here.

https://stackoverflow.com/questions/9294949/when-should-i-use-write-error-vs-throw-terminating-vs-non-terminating-errors has some more on this.